src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/UnadjustedReleaseDate.kt

```diff
@@ -0,0 +1,176 @@
+    var daysToRelease = 0
+    var numberOfDaysToParoleEligibilityDate: Long? = null
+    sentencesInCalculationOrder.forEach { sentencesWithMultipliers ->
+      if (sentencesWithMultipliers.isNotEmpty()) {
+        val releaseStartDate = if (notionalCrd != null) notionalCrd!!.plusDays(1) else sentence.sentencedAt
+        val daysInThisCustodialDuration = SentenceAggregator().getDaysInGroup(releaseStartDate, sentencesWithMultipliers.map { it.sentence }) { it.custodialDuration() }
+        if (sentencesWithMultipliers == sentencesWithPed && !sentence.isRecall()) {
+          numberOfDaysToParoleEligibilityDate = calculateConsecutivePed(sentencesWithMultipliers.map { it.sentence }, daysToRelease, releaseStartDate)
+        }
+        val multiplier = sentencesWithMultipliers[0].multiplier
+        val daysToReleaseInThisGroup = ceil(daysInThisCustodialDuration.toDouble().times(multiplier)).toLong()
+        notionalCrd = releaseStartDate
+          .plusDays(daysToReleaseInThisGroup)
+          .minusDays(1)
+        daysToRelease += daysToReleaseInThisGroup.toInt()
+      }
+    }
+
+    if (sentence.isRecall()) {
+      numberOfDaysToParoleEligibilityDate = null
+    }
+
+    return ReleaseDateCalculation(
+      daysToExpiry,
+      daysToRelease.toDouble(),
+      daysToRelease,
+      numberOfDaysToParoleEligibilityDate,
+    )
+  }
+
+  private fun calculateConsecutivePed(calculableSentences: List<CalculableSentence>, daysToRelease: Int, releaseStartDate: LocalDate): Long {
+    val firstSentence = calculableSentences[0]
+    val firstSentenceMultiplier = determinePedMultiplier(firstSentence.identificationTrack)
+    val sentencesOfFirstType = calculableSentences.filter { determinePedMultiplier(it.identificationTrack) == firstSentenceMultiplier }
+    val sentencesOfOtherType = calculableSentences.filter { determinePedMultiplier(it.identificationTrack) != firstSentenceMultiplier }
+
+    val daysInFirstCustodialDuration = SentenceAggregator().getDaysInGroup(releaseStartDate, sentencesOfFirstType) { it.custodialDuration() }
+    var daysToPed = ceil(daysInFirstCustodialDuration.times(firstSentenceMultiplier)).toLong()
+    val notionalPed = releaseStartDate
+      .plusDays(daysToPed)
+      .minusDays(1)
+    if (sentencesOfOtherType.isNotEmpty()) {
+      val otherSentenceMultiplier = determinePedMultiplier(sentencesOfOtherType[0].identificationTrack)
+      val daysInOtherCustodialDuration = SentenceAggregator().getDaysInGroup(notionalPed, sentencesOfOtherType) { it.custodialDuration() }
+      daysToPed += ceil(daysInOtherCustodialDuration.times(otherSentenceMultiplier)).toLong()
+    }
+    return daysToRelease + daysToPed
+  }
+
+  private fun determinePedMultiplier(identification: SentenceIdentificationTrack): Double {
+    return when (identification) {
+      SentenceIdentificationTrack.SOPC_PED_AT_TWO_THIRDS,
+      SentenceIdentificationTrack.EDS_DISCRETIONARY_RELEASE,
+      -> 2 / 3.toDouble()
```

Comment:
Should we perhaps externalise the 2/3 and 1/2 in configuration?

Reply:
This is existing code thats been lifted from SentenceCalculationService. I'll raise a ticket to clean up this class.